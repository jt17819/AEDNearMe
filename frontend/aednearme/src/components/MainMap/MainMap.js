import React, { useState, useEffect } from 'react'
import { GoogleMap, useLoadScript, Marker, InfoWindow, LoadScript } from "@react-google-maps/api";
import mapStyles from './mapStyles';
import axios from 'axios'
import './MainMap.css'


const MainMap = () => {
  const baseUrl = "http://127.0.0.1:8000"

  const [aedData, setAedData] = useState([])
  const [newLat, setNewLat] = useState();
  const [newLng, setNewLng] = useState();
  const [currentLat, setCurrentLat] = useState();
  const [currentLong, setCurrentLong] = useState();

  useEffect(() => {
    try{
      axios.get(baseUrl + '/aed/')
      .then((res) => {
        setAedData(res.data.data)
      })
    } catch (error) {
      console.log(error)
    }
  }, [])
  
  const containerStyle = {
    width: '100%',
    height: '70vh',
};

  // Hard coded center at Futureproof
  const center = {
      lat: newLat ? newLat : currentLat,
      lng: newLng ? newLng : currentLong
  };

  const options = {
    styles: mapStyles,
    zoomControl: true
  };


if('geolocation' in navigator){
  navigator.geolocation.getCurrentPosition(position => {
    const currentLat = position.coords.latitude;
    setCurrentLat(currentLat)
    const currentLong = position.coords.longitude;
    setCurrentLong(currentLong)
  })
} else {
  console.log('geolocation not available')
}

const position = aedData.map((aed, index) => {
    return {
      lat: parseFloat(aed.lat),
      long: parseFloat(aed.long),
      photo_url: aed.photo_url,
      address: aed.address,
      postCode: aed.post_code,
      what3words: aed.what3words_link
    }
})
  
// const aedInfo = (lat, long, key) => {
//     console.log(lat, long);
//     return <InfoWindow
//       key={key}
//       position={{lat: lat, lng: long}}>
//       <h2>Hello</h2>
//     </InfoWindow>
// }

const [selectedAED, setSelectedAED] = useState(null);

  return (
    <LoadScript googleMapsApiKey="AIzaSyATeYFTD2ha1aawscSrtZxJfJ3m89DB_JU">
        <GoogleMap 
            mapContainerStyle={containerStyle} 
            zoom={15} 
            center={center} 
            options={options}>
          {position.map((aed) => {
            return <Marker 
              key={aed.id}
              position={{lat: aed.lat, lng: aed.long}}
              onClick={() => {
                setSelectedAED(aed);
              }}/>
          })}
       
        {selectedAED && (
          <InfoWindow
            position={{lat: selectedAED.lat, lng: selectedAED.long}}
            onCloseClick={() => {
              setSelectedAED(null)
              setNewLat(selectedAED.lat)
              setNewLng(selectedAED.long)
            }}>
            <div id="info-window">
            <img id="info-image" width={"200px"} src={selectedAED.photo_url}/>
              <ul>
                <li>Address: {selectedAED.address}</li>
                <li>Latitude: {selectedAED.lat}</li>
                <li>Longitude: {selectedAED.long}</li>
                <li>What 3 words: <a href={`${selectedAED.what3words}`} target="_blank">Here</a></li>
              </ul>
            </div>
          </InfoWindow>
        )}
        
        </GoogleMap>

        
    </LoadScript>
  )
}

export default MainMap