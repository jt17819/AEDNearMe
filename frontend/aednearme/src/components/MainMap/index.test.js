import { default as MainMap } from './MainMap'
import { screen, render, fireEvent, waitFor } from '@testing-library/react';
import * as ReactRouterDom from 'react-router-dom';
import { MemoryRouter, Router as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import jwt_decode from 'jwt-decode';
import axios from 'axios'

jest.mock("axios");


describe('Main Map', () => {

    it('renders Loading...', () => {
        render(<MainMap />, { wrapper: MemoryRouter })
        expect(screen.getByText('Loading...')).toBeInTheDocument();
    });

    // test('it renders the page', () => {
    //     render(<MainMap />, { wrapper: MemoryRouter });
    //     const div = screen.getByRole("div");
    //     expect(div).toBeInTheDocument()
    // });

    // test('it renders heading', () => {
    //     render(<MainMap />, {wrapper: MemoryRouter})
    //     const login = screen.getAllByRole('heading');
    //     expect(login[0]).toBeInTheDocument();
    // });

    // test('it lets you put in a username', () => {
    //     render(<MainMap />, {wrapper: MemoryRouter})

    //     const username = screen.getByLabelText('Username');
    //     userEvent.type(username)
    //     // expect(authUser).toHaveBeenCalled();
    // });

    it('should take a snapshot', () => {
        render(<MainMap />, { wrapper: MemoryRouter })
        const { asFragment } = render(<MainMap />, { wrapper: MemoryRouter })
        
        expect(asFragment(<MainMap />)).toMatchSnapshot()
      })

});
