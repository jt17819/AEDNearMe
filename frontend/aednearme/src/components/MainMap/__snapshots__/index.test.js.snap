// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Map should take a snapshot 1`] = `
<DocumentFragment>
  <div />
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
